app/lib/publicLink.ts

```diff
@@ -142,7 +142,8 @@ async function createWasPublicLinkIfAvailable(
 }
 
 export async function unshareCredential(rawCredentialRecord: CredentialRecordRaw): Promise<void> {
-  const vcId = `${credentialIdFor(rawCredentialRecord)}::${rawCredentialRecord.profileRecordId.toHexString?.() || rawCredentialRecord.profileRecordId}`;
+  // Use the unique database record ID to match the key used in createPublicLinkFor
+  const vcId = `${rawCredentialRecord._id.toHexString?.() || rawCredentialRecord._id}::${rawCredentialRecord.profileRecordId.toHexString?.() || rawCredentialRecord.profileRecordId}`;
```

Comment:
Extract this to its own function. Use [`credentialIdFor()`](https://github.com/openwallet-foundation-labs/learner-credential-wallet/blob/main/app/lib/decode.ts#L74), and replace its contents with this logic.